fdbserver/ConsistencyScan.actor.cpp

```diff
@@ -447,16 +456,47 @@ ACTOR Future<Void> checkDataConsistency(Database cx,
 		// If the destStorageServers is non-empty, then this shard is being relocated
 		state bool isRelocating = destStorageServers.size() > 0;
 
+		state int customReplication = configuration.storageTeamSize;
+		if (g_network->isSimulated() && SERVER_KNOBS->DD_MAXIMUM_LARGE_TEAMS > 0 &&
+		    !SERVER_KNOBS->SHARD_ENCODE_LOCATION_METADATA) {
+			for (auto& it : g_simulator->customReplicas) {
+				if (range.begin < std::get<1>(it) && std::get<0>(it) < range.end) {
+					TraceEvent("ConsistencyCheck_CheckCustomReplica")
+					    .detail("ShardBegin", printable(range.begin))
+					    .detail("ShardEnd", printable(range.end))
+					    .detail("SourceTeamSize", sourceStorageServers.size())
+					    .detail("DestServerSize", destStorageServers.size())
+					    .detail("ConfigStorageTeamSize", configuration.storageTeamSize)
+					    .detail("CustomBegin", std::get<0>(it))
+					    .detail("CustomEnd", std::get<1>(it))
+					    .detail("CustomReplicas", std::get<2>(it))
+					    .detail("UsableRegions", configuration.usableRegions)
+					    .detail("First", firstClient)
+					    .detail("Perform", performQuiescentChecks);
+					if (range.begin < std::get<0>(it) || range.end > std::get<1>(it)) {
```

Comment:
Following my above codes, this would be
```
if (!tmpReplicatedRange.contains(range)) {
...
}
```
Why would the other intersection cases be a test failure here? Mathematically, there are 4 possible ways of intersection, all the other 3 ways would lead to a failure test.